### ISSUE TITLE
"--restore-from=" with --pgbackrest-storage-type="s3" does not work

### ISSUE THREAD

**Issue by yuanlinios** [CrunchyData/postgres-operator#1769]
**Which example are you working with?**
I am trying to create a new cluster with "--restore-from", the source cluster has --pgbackrest-storage-type="s3"
```
# the source cluster
pgo create cluster hacluster --replica-count=1 --pgbackrest-storage-type="s3" --pgbouncer -n proj-1
pgo backup hacluster -n proj-1 --pgbackrest-storage-type="s3"

# create the destination cluster
pgo create cluster ha01 --restore-from=hacluster --pgbackrest-storage-type="s3" -n proj-1
```

**What is the current behavior?**
bootstrap pod failed in Error
```
2020-08-09 13:16:30,200 INFO: No PostgreSQL configuration items changed, nothing to reload.
2020-08-09 13:16:30,202 INFO: Lock owner: None; I am ha01-bootstrap-gp9d8
2020-08-09 13:16:30,310 INFO: trying to bootstrap a new cluster
2020-08-09 13:16:30,314 INFO: Running custom bootstrap script: /opt/cpm/bin/pgbackrest/pgbackrest-create-replica.sh primary
Sun Aug  9 13:16:30 UTC 2020 INFO: Empty PGDATA dir found for primary, a non-delta restore will be peformed
ERROR: [055]: unable to load info file '/backrestrepo/hacluster-backrest-shared-repo/backup/db/backup.info' or '/backrestrepo/hacluster-backrest-shared-repo/backup/db/backup.info.copy':
       FileMissingError: raised from remote-0 protocol on 'hacluster-backrest-shared-repo': unable to open missing file '/backrestrepo/hacluster-backrest-shared-repo/backup/db/backup.info' for read
       FileMissingError: raised from remote-0 protocol on 'hacluster-backrest-shared-repo': unable to open missing file '/backrestrepo/hacluster-backrest-shared-repo/backup/db/backup.info.copy' for read
       HINT: backup.info cannot be opened and is required to perform a backup.
       HINT: has a stanza-create been performed?
Sun Aug  9 13:16:30 UTC 2020 ERROR: pgBackRest primary Creation: pgBackRest restore failed when creating primary
2020-08-09 13:16:30,602 INFO: removing initialize key after failed attempt to bootstrap the cluster
2020-08-09 13:16:30,650 INFO: renaming data directory to /pgdata/ha01_2020-08-09-13-16-30
2020-08-09 13:16:30,702 INFO: Lock owner: None; I am ha01-bootstrap-gp9d8
Traceback (most recent call last):
  File "/usr/local/bin/patroni", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 235, in main
    return patroni_main()
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 199, in patroni_main
    patroni.run()
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 135, in run
    logger.info(self.ha.run_cycle())
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1370, in run_cycle
    info = self._run_cycle()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1277, in _run_cycle
    return self.post_bootstrap()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1173, in post_bootstrap
    self.cancel_initialization()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1168, in cancel_initialization
    raise PatroniException('Failed to bootstrap cluster')
patroni.exceptions.PatroniException: 'Failed to bootstrap cluster'
```

**What is the expected behavior?**
The new cluster should be created successfully

**Other information** (e.g. detailed explanation, related issues,  etc)
The bootstrap pod is looking for `/backrestrepo/hacluster-backrest-shared-repo/backup/db/backup.info`. But for a "s3" only config, there is no such file in pgbackrest repo pod. Did I miss any configuration options?

**Please tell us about your environment:**

* Operating System: Centos 7
* Where is this running ( Local , Cloud Provider): Local
* Storage being used (NFS, Hostpath, Gluster, etc): Ceph RBD
* Container Image Tag: 4.4.0
* PostgreSQL Version: 12.3
* Platform (Docker, Kubernetes, OpenShift): Openshift
* Platform Version: 3.11

**Comment by jkatz** [comment]
Did you confirm that the backups for `hacluster` completed?

**Comment by yuanlinios** [comment]
Hello @jkatz 

This is a PoC cluster and has only a few data.


```
pgo create cluster hacluster --replica-count=1 --pgbackrest-storage-type="s3" --pgbouncer -n proj-1
# insert a few rows manually and run some backups ...
pgo backup hacluster -n proj-1 --pgbackrest-storage-type="s3"
```

output from `pgo show backup hacluster -n proj-1`
```
cluster: hacluster
storage type: s3

stanza: db
    status: ok
    cipher: none

    db (current)
        wal archive min/max (12-1)

        full backup: 20200811-125818F
            timestamp start/stop: 2020-08-11 20:58:18 +0800 CST / 2020-08-11 20:58:30 +0800 CST
            wal start/stop: 000000010000000000000002 / 000000010000000000000002
            database size: 31.1MiB, backup size: 31.1MiB
            repository size: 3.7MiB, repository backup size: 3.7MiB
            backup reference list:

        incr backup: 20200811-125818F_20200811-131027I
            timestamp start/stop: 2020-08-11 21:10:27 +0800 CST / 2020-08-11 21:10:32 +0800 CST
            wal start/stop: 000000010000000000000006 / 000000010000000000000007
            database size: 31.1MiB, backup size: 2.2MiB
            repository size: 3.7MiB, repository backup size: 242.2KiB
            backup reference list: 20200811-125818F

        incr backup: 20200811-125818F_20200811-131228I
            timestamp start/stop: 2020-08-11 21:12:28 +0800 CST / 2020-08-11 21:12:31 +0800 CST
            wal start/stop: 000000010000000000000009 / 000000010000000000000009
            database size: 31.1MiB, backup size: 24.3KiB
            repository size: 3.7MiB, repository backup size: 628.0B
            backup reference list: 20200811-125818F, 20200811-125818F_20200811-131027I

        full backup: 20200811-131513F
            timestamp start/stop: 2020-08-11 21:15:13 +0800 CST / 2020-08-11 21:15:21 +0800 CST
            wal start/stop: 00000002000000000000000B / 00000002000000000000000B
            database size: 31.1MiB, backup size: 31.1MiB
            repository size: 3.7MiB, repository backup size: 3.7MiB
            backup reference list:

        incr backup: 20200811-131513F_20200811-131712I
            timestamp start/stop: 2020-08-11 21:17:12 +0800 CST / 2020-08-11 21:17:14 +0800 CST
            wal start/stop: 00000002000000000000000D / 00000002000000000000000D
            database size: 31.1MiB, backup size: 8.3KiB
            repository size: 3.7MiB, repository backup size: 430.0B
            backup reference list: 20200811-131513F
```


I can run ` pgo restore hacluster --pgbackrest-storage-type='s3' -n proj-1` to restore my cluster in-place successfully. 

But create cluster with `--restore-from` like `pgo create cluster ha01 --restore-from=hacluster --pgbackrest-storage-type="s3" -n proj-1` always fail in bootstrap.


Feel free to let me know if you need more info

**Comment by jkatz** [comment]
Please be sure to include `'--repo-type='s3'` in `--restore-opts`, e.g.

```
pgo create cluster ha01 --restore-from=hacluster --restore-opts='--repo-type='s3' --pgbackrest-storage-type="s3" -n proj-1
```

However, there is an issue with default URI style that may be affecting this as well -- this is fixed in #1770. The workaround until that patch is committed is to set a default URI style in the `pgo.yaml` section of `pgo-config`, which will work for newly clusters.
